node/sc/sub_bridge_handler.go

```diff
@@ -361,18 +362,9 @@ func (sbh *SubBridgeHandler) broadcastServiceChainReceiptRequest() {
 
 // updateTxCount update txCount to insert into anchoring tx. It skips first remnant txCount.
 func (sbh *SubBridgeHandler) updateTxCount(block *types.Block) {
-	if sbh.txCountEnabledBlockNumber == 0 {
+	if sbh.txCountStartingBlockNumber == 0 {
 		sbh.txCount = 0 // reset for the next anchoring period
-		sbh.txCountEnabledBlockNumber = block.NumberU64()
-		if sbh.chainTxPeriod > 1 {
-			remnant := block.NumberU64() % sbh.chainTxPeriod
-			if remnant < 2 {
-				// A small trick to start tx counting quickly.
-				sbh.txCountEnabledBlockNumber += 1 - remnant
-			} else {
-				sbh.txCountEnabledBlockNumber += (sbh.chainTxPeriod - remnant) + 1
-			}
-		}
+		sbh.txCountStartingBlockNumber = block.NumberU64()
```

Comment:
How about moving this if branch to `SetAnchoringTx` function?

Reply:
SetAnchoringTx() is called by API. I think that it is not a good place to reset values.